ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutorTest.java

```diff
@@ -0,0 +1,387 @@
+import org.apache.kafka.streams.KafkaClientSupplier;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class InsertValuesExecutorTest {
+
+  private static final Schema SCHEMA = SchemaBuilder.struct()
+      .field("ROWTIME", Schema.OPTIONAL_INT64_SCHEMA)
+      .field("ROWKEY", Schema.OPTIONAL_INT64_SCHEMA)
+      .field("COL0", Schema.OPTIONAL_INT64_SCHEMA)
+      .field("COL1", Schema.OPTIONAL_STRING_SCHEMA)
+      .build();
+
+  private static final Schema STRICT_SCHEMA = SchemaBuilder.struct()
+      .field("ROWTIME", Schema.INT64_SCHEMA)
+      .field("ROWKEY", Schema.INT64_SCHEMA)
+      .field("COL0", Schema.OPTIONAL_INT64_SCHEMA)
+      .field("COL1", Schema.OPTIONAL_STRING_SCHEMA)
+      .build();
+
+  private static final byte[] KEY = new byte[]{1};
+  private static final byte[] VALUE = new byte[]{2};
+
+  private static final String TOPIC_NAME = "topic";
+
+  @Rule
+  public final ExpectedException expectedException = ExpectedException.none();
+
+  @Mock
+  private KsqlEngine engine;
+  @Mock
+  private KsqlTopicSerDe topicSerDe;
+  @Mock
+  private Serde<String> keySerDe;
+  @Mock
+  private Serializer<String> keySerializer;
+  @Mock
+  private Serde<GenericRow> rowSerDe;
+  @Mock
+  private Serializer<GenericRow> rowSerializer;
+  @Mock
+  private ServiceContext serviceContext;
+  @Mock
+  private KafkaProducer<byte[], byte[]> producer;
+
+  @Before
+  public void setup() {
+    when(topicSerDe.getGenericRowSerde(any(), any(), any(), any(), any())).thenReturn(rowSerDe);
+
+    when(keySerDe.serializer()).thenReturn(keySerializer);
+    when(rowSerDe.serializer()).thenReturn(rowSerializer);
+
+    when(keySerializer.serialize(any(), any())).thenReturn(KEY);
+    when(rowSerializer.serialize(any(), any())).thenReturn(VALUE);
+
+    final KafkaClientSupplier kafkaClientSupplier = mock(KafkaClientSupplier.class);
+    when(kafkaClientSupplier.getProducer(any())).thenReturn(producer);
+
+    when(serviceContext.getKafkaClientSupplier()).thenReturn(kafkaClientSupplier);
+
+    givenDataSourceWithSchema(SCHEMA);
+  }
+
+  @Test
+  public void shouldHandleFullRow() {
+    // Given:
+    final ConfiguredStatement<InsertValues> statement = givenInsertValues(
+        SCHEMA.fields().stream().map(Field::name).collect(Collectors.toList()),
+        ImmutableList.of(
+            new LongLiteral(1L),
+            new LongLiteral(2L),
+            new LongLiteral(2L),
+            new StringLiteral("str"))
+    );
+
+    // When:
+    new InsertValuesExecutor(() -> 1L).execute(statement, engine, serviceContext);
+
+    // Then:
+    verify(rowSerializer).serialize(TOPIC_NAME, new GenericRow(1L, 2L, 2L, "str"));
+    verify(producer).send(new ProducerRecord<>(TOPIC_NAME, KEY, VALUE));
```

Comment:
Doesn't this need to include the timestamp provided in `InsertValues`?

Reply:
Oops I updated the code and forgot to re-run the tests!